### ISSUE TITLE
`seq` cannot produce quote-needing results

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1479]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2086
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

# init-nfsmachine - repopulate known_hosts interactively

for i in $(seq 1 8)
do
    ssh foo${i}.bar.com "echo 'Connected!'"
done

```

#### Here's what shellcheck currently says:
```
Line 7	SC2086: Double quote to prevent globbing and word splitting.
```
#### Here's what I wanted or expected to see:

Nothing. `seq` prints numbers separated by newlines. Each number cannot contain spaces, therefore this `i` does not need quotes. 

If the shellcheck analysis cannot determine that the `i`, unassigned between its initial assignment and this usages, will not contain spaces. then feel free to close.

**Comment by michaelsstuff** [comment]
Similar issue would be: 
```bash
for i in $(seq 3 $END); do
    echo "$i"
done
```

![image](https://user-images.githubusercontent.com/11227197/111616560-51394b00-87e2-11eb-91c2-41d9015b31cb.png)

If you add quotes to the END, seq throws an error:
```
seq: invalid floating point argument: ‘’
Try 'seq --help' for more information.
```
